tests/unit/test_uat.py

```diff
@@ -0,0 +1,70 @@
+# package imports
+from unittest import mock
+
+import responses
+from earthaccess import Auth, search_data
+from earthaccess.auth import CLIENT_ID, Env
+
+
+class TestUatEnvironmentArgument:
+    @responses.activate  # This will cause the test to check that all mocked URLs are hit.
+    @mock.patch("getpass.getpass", new=mock.Mock(return_value="password"))
+    @mock.patch(
+        "builtins.input",
+        new=mock.Mock(return_value="user"),
+    )
+    def test_uat_login_when_uat_selected(self) -> bool:
+        """Test the correct env is queried based on what's selected at login-time."""
+        json_response = [
+            {"access_token": "EDL-token-1", "expiration_date": "12/15/2021"},
+            {"access_token": "EDL-token-2", "expiration_date": "12/16/2021"},
+        ]
+        responses.add(
+            responses.GET,
+            "https://uat.urs.earthdata.nasa.gov/api/users/tokens",
+            json=json_response,
+            status=200,
+        )
+        responses.add(
+            responses.GET,
+            "https://uat.urs.earthdata.nasa.gov/profile",
+            json={"email_address": "test@test.edu"},
+            status=200,
+        )
+        responses.add(
+            responses.GET,
+            f"https://uat.urs.earthdata.nasa.gov/api/users/user?client_id={CLIENT_ID}",
+            json={},
+            status=200,
+        )
+
+        # TODO: Add a mock for CMR query?  Does it need a "CMR-HITS" field in the response?
```

Comment:
When the following 6 lines are uncommented, then the `results = search_data(session=session)` line fails because `CMR-HITS` is expected to be in the response:

```
>       results = search_data(session=session)

test_uat.py:68: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../earthaccess/api.py:127: in search_data
    granules_found = query.hits()
../../earthaccess/search.py:435: in hits
    return int(response.headers["CMR-Hits"])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = {'Content-Type': 'application/json'}, key = 'CMR-Hits'

    def __getitem__(self, key):
>       return self._store[key.lower()][1]
E       KeyError: 'cmr-hits'
```